### PR TITLE
[IDB Ignore] Icons folder reorganization and cleaning

### PR DESCRIPTION
**Pull request by Blundir**
# About the pull request

Splits and reorganizes tons of heavy dmi files, removes some completely unused icons.

No changes to existing icons, no new ones are added.

No player facing changes. (Apart from fixes)

- [x] Split majority of heavy files into smaller files with better structure
- [x] Attach new icon paths based on icon_state (aka object icon)
- [x] Attach new icon paths based on item_state (aka mob icon)
- [x] Add support for camo items render in vendors
- [x] Rework `select_gamemode_skin` into not forcing icon prefixes like b_ c_ u_ on everything it touches
- [x] Rework helmet garb

# Explain why it's good for the game

We hit icon state cap (512) in certain files, and not to mention the amount of merge conflicts overloaded files have.

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl: Blundir, Drathek, ItsVyzo, CzHa, forest2001
refactor: reorganized and cleaned many dmis in icons folder
del: inflatable duck can no longer have ammo
del: removed a huge dozen of extremely outdated legacy bay items
imagedel: removed majority of left unused, duplicate and legacy sprites from the codebase
imageadd: es7 in belt holstered sprite
imageadd: m39 pmc in belt holstered sprite
imageadd: warden's uniform now has camo variants
imageadd: dcc's jacket now has camo variants
imageadd: type 71, type 71-F and type 71 commando suit storage icons by CzHa (discord tag .cpol)
fix: mp uniforms not having working onmob camo sprites
fix: mob looking handcuffed while just holding cuffs in hand
fix: rank pins/boards and other accessories not being visible on object sprite
fix: pdt bracelet looking like a blue tie on the uniform
fix: officer uniform not having working camo sprites
fix: m39 not having custom sprites on belt for camo variants
fix: taser having positions in hands wrong
fix: inhands for shotgun shells not working
fix: emp grenades now use correct (non legacy) icon 
/:cl:

### CODE REVIEW

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -262,18 +266,36 @@
...
+				if("desert")
+					icon_state = new_icon_state ? new_icon_state : "d_" + icon_state
+					item_state = new_item_state ? new_item_state : "d_" + item_state
+				if("classic")
+					icon_state = new_icon_state ? new_icon_state : "c_" + icon_state
+					item_state = new_item_state ? new_item_state : "c_" + item_state
+		if(new_protection)
+			min_cold_protection_temperature = new_protection
+		else
+			switch(SSmapping.configs[GROUND_MAP].camouflage_type)
```
Comment (Drulikar) [inline code/game/objects/items.dm:283]:
```suggestion
			if(!item_icons)
				item_icons = list()
			switch(SSmapping.configs[GROUND_MAP].camouflage_type)
```
See if this resolves your bad index issues.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Review by harryob — APPROVED** [review]
reviewed commits since 7fd61825544a0684791ff2417bc0244c362398bc, lgtm

**Pull request merged**